DS/assignmentds4/HashTable.java

```diff
@@ -0,0 +1,112 @@
+package assignmentds4;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/*	Assignment DS4 Q2.
+ * 	Write a program to allot rooms in guest house. rooms can be identified by their room number and
+ *	allotment is based age of Guest. (Use Hash Table techniques for allocation)
+ *	Given : There are N numbers of rooms and K is the count of guests,
+ *	where N is a prime number and N > K
+ *
+ *	customer age is used as key and for key collision.. preventive measures has been coded
+ */
+public class HashTable {
+    private int[] rooms;
+    private Customer[] customers;
+    private int occupiedRooms;
+    private PrimeGenerator primeGenerator;// key generator for hashtable
+    private final int N = 51; // TOTAL_NUMBER_OF_ROOMS
+
+    public HashTable() {
+	rooms = new int[N];
+	customers = new Customer[50];
+	occupiedRooms = 0;
+	primeGenerator = new PrimeGenerator();
+    }
+
+    // will set any data value on location of key
+    public void put(int key, Customer customer) {
+	if (isExiting(key) == false) {
+	    rooms[occupiedRooms++] = key;
+	    customers[occupiedRooms - 1] = customer;
```

Comment:
occupiedRooms will increase while update the customers array. No need to update room quantity in rooms array.